### PR TITLE
Relax types on `DigestItemRef`, such that byte slices can be used in addition to vector references

### PR DESCRIPTION
**Pull request by nazar-pc**
The goal here is to use `DigestItemRef` with `frame_support::traits::FindAuthor`, so I can turn `(ConsensusEngineId, &'a [u8])` into `DigestItemRef` without extra allocation:
```
pub trait FindAuthor<Author> {
	/// Find the author of a block based on the pre-runtime digests.
	fn find_author<'a, I>(digests: I) -> Option<Author>
	where
		I: 'a + IntoIterator<Item = (ConsensusEngineId, &'a [u8])>;
}
```

Ideally, I'd like to see `FindAuthor` having `&'a DigestItem` instead of `(ConsensusEngineId, &'a [u8])` if that is OK (this PR is still useful though).

On a similar note `DigestItemRef` should really use `ConsensusEngineId` instead of `&'a ConsensusEngineId` because reference will be more expensive to copy than the data itself.

Technically this PR introduces a breaking change to the API, but it is extremely unlikely to be an issue in practice as the source of that iterator is almost certainly `Vec<DigestItem>`.

### CODE REVIEW

File: primitives/runtime/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "sp-runtime"
-version = "4.0.0"
+version = "4.1.0-dev"
```
Comment (bkchr) [inline primitives/runtime/Cargo.toml:3]: Why did you bumped the version? 

Based on your change this is not correct anyway. Could you please revert this.

**Review by bkchr — APPROVED** [review]
Looks good, beside the version bump.

File: primitives/runtime/Cargo.toml (hunk 1 above)
Comment (nazar-pc) [reply primitives/runtime/Cargo.toml:3]: Leaving `4.0.0` is incorrect either, the code has changed and the version in the codebase differs from what is on `crates.io`. I believe I saw similar changes for other crates recently when changes were introduced to already published crates.

Comment (bkchr) [reply primitives/runtime/Cargo.toml:3]: Yes, but then you would need to bump it to 5.0.0

Comment (nazar-pc) [reply primitives/runtime/Cargo.toml:3]: If this is considered a breaking change - yes. I can bump to `5.0.0-dev` if that makes more sense, I don't see specific guidelines for that in contribution doc.

Comment (nazar-pc) [reply primitives/runtime/Cargo.toml:3]: Similar bump was done in #10430 for instance. Let me know if you still want to revert version change here so we can move forward with this.

Comment (bkchr) [reply primitives/runtime/Cargo.toml:3]: Merge master and then we can continue.

**Comment by bkchr** [comment]
bot merge

**Comment by bkchr** [comment]
@nazar-pc Can you add a polkadot companion?

**Comment by nazar-pc** [comment]
This PR is meant to be fully compatible with it and indeed I see no issues locally :man_shrugging: 
CI errors do not seem related to this PR, I pulled `master` again just in case.

**Comment by bkchr** [comment]
This didn't helped.

**Comment by nazar-pc** [comment]
I think there is something related to lock file that forces it to load incorrect version of the package whose version was updated, but I can't reproduce the issue locally.

**Pull request merged**